src/search/hybridRetriever.ts

```diff
@@ -2,28 +2,21 @@ import { BrevilabsClient } from "@/LLMProviders/brevilabsClient";
 import { extractNoteTitles, getNoteFileFromTitle } from "@/utils";
 import { BaseCallbackConfig } from "@langchain/core/callbacks/manager";
 import { Document } from "@langchain/core/documents";
-import { Embeddings } from "@langchain/core/embeddings";
-import { BaseLanguageModel } from "@langchain/core/language_models/base";
 import { ChatPromptTemplate } from "@langchain/core/prompts";
 import { BaseRetriever } from "@langchain/core/retrievers";
 import { search } from "@orama/orama";
-import { Vault } from "obsidian";
 import { DBOperations } from "./dbOperations";
+import VectorStoreManager from "@/search/vectorStoreManager";
+import ChatModelManager from "@/LLMProviders/chatModelManager";
+import EmbeddingManager from "@/LLMProviders/embeddingManager";
+import { getSettings } from "@/settings/model";
 
 export class HybridRetriever extends BaseRetriever {
   public lc_namespace = ["hybrid_retriever"];
 
-  private llm: BaseLanguageModel;
   private queryRewritePrompt: ChatPromptTemplate;
-  private embeddingsInstance: Embeddings;
-  private brevilabsClient: BrevilabsClient;
 
   constructor(
-    private dbOps: DBOperations,
```

Comment:
hybridRetriever becomes a lot easier to use after taking out these singleton instance args